datafusion/src/physical_plan/functions.rs

```diff
@@ -1373,20 +1374,28 @@ impl PhysicalExpr for ScalarFunctionExpr {
     }
 
     fn evaluate(&self, batch: &RecordBatch) -> Result<ColumnarValue> {
```

Comment:
I think this approach, while a hack, seems reasonable to me

I agree with the comments on https://github.com/apache/arrow-datafusion/pull/303#discussion_r629859725 that this does seem like a hack (it would be nicer to have an enum or something that made this case explicit).


I would like to see this expectation documented near the place where a user would define a UDF as well - either as an example or as a doccomment. Perhaps somewhere in 
https://github.com/apache/arrow-datafusion/blob/master/datafusion/src/physical_plan/udf.rs

Or bonus points for adding an example:
https://github.com/apache/arrow-datafusion/blob/master/datafusion-examples/examples/simple_udf.rs


cc; @msathis  who I think faced a similar challenge in `now()`: https://github.com/apache/arrow-datafusion/pull/288

Reply:
@alamb tried a different approach using `TryFrom` trait, so it's more obvious. the call site convention I'll try to code in #303

Reply:
> Docs in the next PR is fine. I think @jorgecarleitao had an alternate suggestion in [#303 (comment)](https://github.com/apache/arrow-datafusion/pull/303#discussion_r630285020)

see https://github.com/apache/arrow-datafusion/pull/328